writing-your-first-es2015-javascript-module.md

```diff
@@ -0,0 +1,140 @@
+<!--
+headline: Writing Your First ES2015 JavaScript Module
+description: dead simple guide to writing your first ES2015 module using Babel.
+author: Ben Coe
+datePublished: 2015-11-29
+twitter: benjamincoe
+github: bcoe
+-->
+
+> ES2015 introduces a many slick language features (arrow functions, classes, template strings), but it
+  can be overwhelming figure out where to begin with it!
+
+The library [nyc](https://github.com/bcoe/nyc) grew out of my frustration with how complicated it was to instrument JavaScript projects with test coverage:
+
+```js
+mocha --ui exports --require blanket -R mocoverage
+```
+
+incantations like this were the norm in my _package.json_'s script stanza.
+
+I dreamt of a simpler world, one where you could write `nyc mocha` and have test coverage work like magic.
+
+_this world turned out to be awesome!_
+
+Recently an [issue](https://github.com/bcoe/nyc/issues/53) was opened on nyc requesting support for Babel.
+This brought my attention to a problem similar to the one that nyc was originally built to solve ...
+
+There are many different approaches folks use to wire up ES2015 to ES5 transpilation, the methods use a variety
+of different build tools (_gulp_, _grunt_, _webpack_, _browserify_), and the paradox of choice can be overwhelming.
+
+In the process of adding Babel support to nyc I took notes, my goal being to outline a
+_dead simple_ approach for writing ES2015 modules that work on older versions of Node.js: using
+npm scripts instead of build tools, keeping library dependencies to a minimum, and avoiding
+build steps where possible.
+
+ES2015 introduces many [awesome language features](https://babeljs.io/docs/learn-es2015/) to your JavaScript repertoire,
+and I hope this tutorial will convince you to dive in and start playing.
+
+For the purpose of this discussion I've created the ES2015 module yarsay, clone it and follow along:
+
+[https://github.com/bcoe/yarsay](https://github.com/bcoe/yarsay)
+
+## Installing and Configuring Babel
+
+To get up and running you'll first want to install the babel development dependencies:
+
+```sh
+npm i babel-cli babel-core babel-preset-es2015 --save-dev
+```
+
+Let's look at each of these:
+
+* *babel-core:* the bare-bones babel compiler (without the plugins that make it actually transpile).
+* *babel-preset-es2015:* provides the set of plugins necessary to transpile ES2015 code.
+* *babel-cli:* the command-line tool used to package your module for publication to npm.
+
+> transpilation refers to converting our ES2015 source code to ES5, a flavor of JavaScript that works on older Node.js platforms.
```

Comment:
[cap] Not sure if this is an actual quote, but I would capitalize "transpilation".

[grammar] I would also use "your ES2015 source" instead of "our ES2015 source".